src/lib/y2network/wicked/connection_config_readers/base.rb

```diff
@@ -158,14 +159,37 @@ def all_ips
         # It takes an IP address and, optionally, a prefix or a netmask.
         #
         # @param ip      [Y2Network::IPAddress] IP address
-        # @param prefix  [Integer,nil] Address prefix
-        # @param netmask [String,nil] Netmask
-        def build_ip(ip, prefix, netmask)
+        # @param id      [String] Hash key for the IP Address
+        def build_ip(ip, id)
           ipaddr = ip.clone
           return ipaddr if ip.prefix?
 
-          ipaddr.netmask = netmask if netmask
-          ipaddr.prefix = prefix if prefix
+          begin
+            netmask = file.netmasks[id]
+            ipaddr.netmask = netmask if netmask
+          rescue StandardError
+            issue_location = "file:#{file.path}:NETMASK#{id}"
+            issue = Y2Issues::InvalidValue.new(
+              netmask, fallback: nil, location: issue_location
+            )
+            issues_list << issue
+          end
+
+          prefix = file.prefixlens[id]
+          return ipaddr unless prefix
+
+          begin
+            address = ipaddr.address.clone
```

Comment:
I do not get why you are cloning the address here. Actually, it looks like you are just using the new `address` variable in the next line, but nothing else.

Reply:
I do not wanted to modify the address at all but only take advantage of the already provided validations as we are not modifying it in other circumstances like when a netmask is assigned.

Reply:
I was also thinking in doing it overriding the `prefix=` method similar to what we do for the netmask.

https://github.com/yast/yast-network/blob/master/src/lib/y2network/ip_address.rb#L87

Reply:
The intention is still not clear to me. A comment would help. Otherwise, if the only line you expect to fail is the `address`, you might put only that line into the begin/rescue block.

Reply:
Well, I do not want the prefix to be assigned in case it is invalid.